kroxylicious-kms-provider-hashicorp-vault-test-support/src/main/java/io/kroxylicious/kms/provider/hashicorp/vault/VaultTestKmsFacade.java

```diff
@@ -40,39 +48,83 @@ public boolean isAvailable() {
 
     @Override
     @SuppressWarnings("resource")
-    public void start() {
-
-        vaultContainer = new VaultContainer<>(HASHICORP_VAULT)
-                .withVaultToken(VAULT_TOKEN)
-                .withEnv("VAULT_FORMAT", "json")
-                .withInitCommand(
-                        "secrets enable transit");
+    public void startVault() {
+        vaultContainer = new VaultContainer<>(HASHICORP_VAULT).withVaultToken(VAULT_ROOT_TOKEN);
         vaultContainer.start();
     }
 
     @Override
-    public void stop() {
+    public void stopVault() {
         if (vaultContainer != null) {
             vaultContainer.close();
         }
     }
 
     @Override
-    public TestKekManager getTestKekManager() {
-        return new VaultTestKekManager();
+    protected void enableTransit() {
+
+        var engine = new EnableEngine("transit");
+
+        var body = getBody(engine);
+
+        var request = createVaultRequest()
+                .uri(getVaultUrl().resolve("v1/sys/mounts/transit"))
+                .POST(HttpRequest.BodyPublishers.ofString(body))
+                .build();
+
+        sendRequestExpectingNoContentResponse(request);
+    }
+
+    @Override
+    protected String createOrphanToken(String description, boolean noDefaultPolicy, Set<String> policies) {
+
+        var token = new CreateTokenRequest(description, noDefaultPolicy, policies);
+
+        String body = getBody(token);
+
+        var request = createVaultRequest()
+                .uri(getVaultUrl().resolve("v1/auth/token/create-orphan"))
+                .POST(HttpRequest.BodyPublishers.ofString(body))
+                .build();
```

Comment:
could extract a method for this like `createVaultPost(String path, Object bodyObj)`